### DISCUSSION TITLE
anti antidebugger help

### DISCUSSION THREAD

**Discussion by peterxiyong** [uBlockO/uBO-Scriptlets#25] [Q&A] [answered]
hii @uBlock-user
please fix antidebugger on these websites(when dev tool is opened, website redirects to home page)
```
https://bflix.io/home
https://bflix.io/movie/the-little-mermaid2-zk9wl/1-1
https://swatchseries.ru/movie/oppenheimer-j2r74/1-1
```

**Comment by uBlock-user** [comment, marked as answer]
@stephenhawk8054

**Reply by peterxiyong** [reply]
any working filters?

**Reply by uBlock-user** [reply]
`bflix.io##+js(set, console, undefined)` works ?

**Reply by peterxiyong** [reply]
it does not work, still it redirects to homepage

**Reply by stephenhawk8054** [reply]
Does this work?

```adb
bflix.io##+js(aost, console, all.js)
```

**Reply by peterxiyong** [reply]
it doesn't work, videos are broken

**Reply by stephenhawk8054** [reply]
How about

```adb
bflix.io##+js(aost, console.clear, all.js)
```
?

**Reply by peterxiyong** [reply]
no it doesn't work...when i go to watch movies & open dev tools, it redirects to homepage

**Reply by stephenhawk8054** [reply]
Hmmm... it doesn't redirect to home page for me: https://github.com/uBlock-user/uBO-Scriptlets/assets/66517106/a59e5532-927f-4d6a-b678-f7fbec40f1db

The iframe `vidstream.pro` has anti-devtools that freezes the video for me tho.

But `bflix.io` doesn't redirect at all for me with those filters.

**Reply by uBlock-user** [reply]
If we can find a fix for this, then the same fix can be applied for fmovies and aniwave websites too.

**Reply by stephenhawk8054** [reply]
These work partially for me:

```adb
bflix.io##+js(aost, console.clear, all.js)
vidstream.pro##+js(aost, console.clear, embed.js)
```

which is if I load the website first -> open devtools then it's ok. If I open the devtools -> load the website then it's frozen.

https://github.com/uBlock-user/uBO-Scriptlets/assets/66517106/3abda78c-2361-41f6-9756-83ad270839a2

**Reply by peterxiyong** [reply]
with above filters
bflix.io=> bflix.io/home => open dev tools => redirects to bflix.io

**Reply by stephenhawk8054** [reply]
Well, I cannot reproduce any more so I don't know how to investigate further, video recording: https://ttm.sh/I2n.mp4

**Reply by peterxiyong** [reply]
click watch now on any movie, open dev tools then redirects to bflix.io

**Reply by stephenhawk8054** [reply]
I can't reproduce either: https://ttm.sh/I2_.mp4

**Reply by peterxiyong** [reply]
did you try google chrome

**Reply by stephenhawk8054** [reply]
Ok, I have no idea you are talking about chromium.

Do these work?

```adb
bflix.io##+js(aost, console.clear, all.js)
bflix.io##+js(nostif, ||null)
```

**Reply by peterxiyong** [reply]
ok now redirection stopped.
vidstream hides source in console tab & also in dom

**Reply by uBlock-user** [reply]
These filters work for me and fix everything - 

```
bflix.io,swatchseries.ru,vidstream.pro,mcloud.to##+js(aost, console.clear)
bflix.io,swatchseries.ru,vidstream.pro,mcloud.to##+js(nostif, 0x)
bflix.io,swatchseries.ru,vidstream.pro,mcloud.to##+js(nosiif, 0x)
```

**Reply by stephenhawk8054** [reply]
I don't think we need `/^/`, just `##+js(aost, console.clear)` is enough for the first filter?

**Reply by uBlock-user** [reply]
Sure, if absense is equivalent to all files ? I haven't looked into `aost` scriptlet.

**Reply by stephenhawk8054** [reply]
Yeah, empty string means [always match](https://github.com/gorhill/uBlock/wiki/Resources-Library#abort-on-stack-tracejs-)

**Reply by uBlock-user** [reply]
ok, modified. Are you going to add these filters to uAssets ?

**Reply by stephenhawk8054** [reply]
Sure: https://github.com/uBlockOrigin/uAssets/commit/2ee02591ab1c907c8f7cdbd62f55a7b4caeae685

**Reply by uBlock-user** [reply]
also add `fmovies.to` and `aniwave.to` @stephenhawk8054

**Reply by stephenhawk8054** [reply]
Added: https://github.com/uBlockOrigin/uAssets/commit/fac90620251633d9265878c728dd048c4eec82e8

**Reply by uBlock-user** [reply]
@stephenhawk8054 `fmovies.to` has changed to `fmoviesz.to`, also add `vidplay.site`

**Reply by stephenhawk8054** [reply]
Added: https://github.com/uBlockOrigin/uAssets/commit/b9a47de7ed3fb5650469eda2333637a6eebda8ba

**Reply by uBlock-user** [reply]
`##+js(nostif, ||null)` doesn't stop the issue and should be removed fyi

**Reply by stephenhawk8054** [reply]
Ok: https://github.com/uBlockOrigin/uAssets/commit/5ba067fd2a44d4c37f71aee89933058db29144f7

**Reply by uBlock-user** [reply]
@stephenhawk8054 issue is back.

**Reply by michmue** [reply]
for me the issue is still fixed.

Has someone a good link/doc on how to create these filters by myself? (@stephenhawk8054)
I know how to use the scriptlets but dont know the right way to find the right parameters ` console.clear` and `0x`


my insights after all.js deobfuscation: 
1. found the a source code on github https://github.com/theajack/disable-devtool
2. found _0x5c7563 (uses jquery as an parameter) stacktrace: runs anonymouse function calling `debugger` when in dev tools
3. does `nostif, 0x` kills a setTimeOut on _0x5c7563 or is 0x meant for something else?

little bit lost :/

4. is deobfuscation needed?

deobfuscated code https://gist.github.com/michmue/ce0f8c7f49b13c4f45732a643ddd151b

**Reply by uBlock-user** [reply]
@michmue For me, its happens when I switch to debugger tab in devtools and a redirection to the homepage from any page on - `https://fmoviesz.to/home`

On `aniwave.to` F12 is blocked and even if you manage to unblock it, redirection to homepage as soon as devtools is open.

**Reply by michmue** [reply]
@uBlock-user Was blocked on aniwave.to (to much redirects or something while in debbuger). fmoviesz.to also blocks me now. Cannot verify for the next 21:14h :D.

`Your IP has been blocked
Because of abused our system, all your requests will be denined until 2023-12-11 09:32 (Remaining: 21:14:44)
Your IP is: ...`

**Reply by uBlock-user** [reply]
@michmue fixed `aniwave.to` with `aniwave.to,vidplay.*##+js(set, DisDevTool, undefined)` but same fix doesn't work for `fmoviesz.to`

**Reply by stephenhawk8054** [reply]
Updated for `aniwave.to`: https://github.com/uBlockOrigin/uAssets/commit/d02406d1e2e8d532b86d0879ea36f182557f28b8

I can't check `fmoviesz.to` right now so feel free to ping me if any solution is found.

**Reply by huyszn** [reply]
@stephenhawk8054 I found `fmoviesz.to##+js(nostif, P())` works on `fmoviesz.to` but there might be a better solution.

**Reply by uBlock-user** [reply]
@stephenhawk8054 Thanks, and @huyszn's filter works as expected when tested on Debugger tab in devtools pane.

---

`fmoviesz.to##+js(nostif, null)` also works

**Reply by stephenhawk8054** [reply]
@huyszn Does `fmoviesz.to##+js(nostif, null)` work on your side?

**Reply by huyszn** [reply]
@stephenhawk8054 `fmoviesz.to##+js(nostif, null)` also works for me.

**Reply by stephenhawk8054** [reply]
Updated: https://github.com/uBlockOrigin/uAssets/commit/6d327720f40b4120252de8a65d37b5d368001e47

**Reply by huyszn** [reply]
Found another site `animesuge.to` that has this same issue. `animesuge.to##+js(nostif, null)` fixes it on chrome and `animesuge.to##+js(set, DisDevTool, undefined)` fixes it on firefox.

**Reply by uBlock-user** [reply]
> fixes it on chrome

`nostif` will not address the F12 key issue, you need a complete set of filters to address all issues together.

```
animesuge.to##+js(nostif, null)
animesuge.to##+js(set, DisDevTool, undefined)
animesuge.to##+js(aost, console.clear)
```

**Reply by stephenhawk8054** [reply]
Updated: https://github.com/uBlockOrigin/uAssets/commit/733195325030ca4bc9c019c09edd57f25ef9aaa9

**Reply by huyszn** [reply]
I think the filters work on `hurawatch.bz` and `hdtoday.so`

**Reply by stephenhawk8054** [reply]
Added: https://github.com/uBlockOrigin/uAssets/commit/c732a6ac6232baaa309796d9b3eb491a8c82f7a9

**Reply by uBlock-user** [reply]
@stephenhawk8054 Change `||null` to `null` for `aniwave.to` in ubo annoyances list

**Reply by uBlock-user** [reply]
>  @stephenhawk8054 Change ||null to null for aniwave.to in ubo annoyances list

@stephenhawk8054

**Reply by stephenhawk8054** [reply]
Oh, I didn't see the above notification. Updated: https://github.com/uBlockOrigin/uAssets/commit/337a07b29e6f6485e88e464f3a8bbd2fc76a68be

**Reply by uBlock-user** [reply]
@stephenhawk8054 change `null` to `||null`, it breaks description cards on the homepages

**Reply by stephenhawk8054** [reply]
https://github.com/uBlockOrigin/uAssets/commit/53d65f123ff4ba6ebd4b8a4d5e6147b366985bb6

**Reply by uBlock-user** [reply]
@stephenhawk8054 `animesuge.to,fmoviesz.to,hdtoday.so,hurawatch.bz##+js(nostif, null)` too

**Reply by stephenhawk8054** [reply]
https://github.com/uBlockOrigin/uAssets/commit/425cbd509b2ceaa80f877a4e8628568b5233185a

**Reply by uBlock-user** [reply]
@stephenhawk8054 domain update for fmovies - `fmovies24.to` 

plz add this asap.

**Reply by stephenhawk8054** [reply]
https://github.com/uBlockOrigin/uAssets/commit/bcca7dc577fb41ebe72f2d3630b79b961a6dc491
